### PR TITLE
fix(steps): fix int column has no `isnan()` method

### LINKED ISSUE #147: bug(step):  HandleUnivariateOutliers
code to reproduce error
```python
import ibis
import ibis_ml as ml

p = ibis.examples.penguins.fetch()
df_train, df_test = ml.train_test_split(p, unique_key="bill_length_mm", test_size=0.3, random_seed=12)
X_train = p.drop(["sex"])
y_train = p['sex'].cases((("male", 1), ("female", 0)), default=0).cast("int64").name("class")
recipe = ml.Recipe(
    ml.HandleUnivariateOutliers(
        ml.numeric(),
        treatment="trimming"
        ),
)
recipe.fit(X_train, y_train)
```

error
```
File ~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_handle_outliers.py:145, in HandleUnivariateOutliers.transform_table(self, table)
    [131](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_handle_outliers.py:131)     return table.mutate(
    [132](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_handle_outliers.py:132)         **{
    [133](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_handle_outliers.py:133)             col_name: table[col_name].clip(
   (...)
    [137](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_handle_outliers.py:137)         }
    [138](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_handle_outliers.py:138)     )
    [139](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_handle_outliers.py:139) else:
    [140](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_handle_outliers.py:140)     return table.filter(
    [141](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_handle_outliers.py:141)         [
    [142](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_handle_outliers.py:142)             (
    [143](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_handle_outliers.py:143)                 (table[col_name] >= stat["lower_bound"])
    [144](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_handle_outliers.py:144)                 & (table[col_name] <= stat["upper_bound"])
--> [145](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_handle_outliers.py:145)                 | (table[col_name].isnull() | table[col_name].isnan())  # noqa: PD003
    [146](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_handle_outliers.py:146)             )
    [147](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_handle_outliers.py:147)             for col_name, stat in self.stats_.items()
    [148](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_handle_outliers.py:148)         ]
    [149](https://file+.vscode-resource.vscode-cdn.net/Users/jiting/repos/~/anaconda3/envs/ibis-ml-dev/lib/python3.12/site-packages/ibis_ml/steps/_handle_outliers.py:149)     )

AttributeError: 'IntegerColumn' object has no attribute 'isnan'
```

### PR DESCRIPTION
**Pull request by jitingxu1**
resolve #147

### CODE REVIEW

File: ibis_ml/steps/_handle_outliers.py [hunk 1]
```diff
@@ -142,7 +142,7 @@ def transform_table(self, table: ir.Table) -> ir.Table:
                     (
                         (table[col_name] >= stat["lower_bound"])
                         & (table[col_name] <= stat["upper_bound"])
-                        | (table[col_name].isnull() | table[col_name].isnan())  # noqa: PD003
+                        | table[col_name].isnull()  # noqa: PD003
```
Comment (jcrist) [inline ibis_ml/steps/_handle_outliers.py:145]: This is not the correct fix. Instead we'll want to branch on `type().is_floating()` and only call `isnan` for floating inputs.

Comment (jitingxu1) [reply ibis_ml/steps/_handle_outliers.py]: Thanks

File: tests/test_handle_outliers.py [hunk 2]
```diff
@@ -1,66 +1,109 @@
...
+def test_handle_univariate_outliers(
+    deviation_factor, method, treatment, cols, test_table, expected
+):
     train_table = ibis.memtable(
         {
             # use same value for easier calculation statistics
-            "col1": [cols["col1"]] * 10,  # mean = 0, std = 0
-            "col2": [cols["col2"]] * 10,  # Q1 = 1, Q3 = 1
+            "int_col": [0] * 10,  # mean = 0, std = 0 Q1 = 0, Q3 = 0
+            "floating_col": [0.0] * 10,  # mean = 0, std = 0 Q1 = 0, Q3 = 0
```
Comment (deepyaman) [inline tests/test_handle_outliers.py:95]:
```suggestion
            "floating_col": [0.0] * 10,  # mean = 0, std = 0, Q1 = 0, Q3 = 0
```

File: ibis_ml/steps/_handle_outliers.py [hunk 3]
```diff
@@ -142,7 +142,11 @@ def transform_table(self, table: ir.Table) -> ir.Table:
                     (
                         (table[col_name] >= stat["lower_bound"])
                         & (table[col_name] <= stat["upper_bound"])
-                        | (table[col_name].isnull() | table[col_name].isnan())  # noqa: PD003
+                        | (
+                            (table[col_name].isnull() | table[col_name].isnan())  # noqa: PD003
+                            if table[col_name].type().is_floating()
+                            else table[col_name].isnull()  # noqa: PD003
+                        )
```
Comment (deepyaman) [inline ibis_ml/steps/_handle_outliers.py:149]:
```suggestion
                        | (table[col_name].isnull() | table[col_name].type().is_floating() & table[col_name].isnan()) # noqa: PD003
```
seems much simpler, unless I'm missing something?

**Review by deepyaman — CHANGES_REQUESTED** [review]

File: ibis_ml/steps/_handle_outliers.py (hunk 3 above)
Comment (jitingxu1) [reply ibis_ml/steps/_handle_outliers.py:149]: Yes, it is simpler.

**Review by deepyaman — APPROVED** [review]
:shipit:

**Pull request merged**